### PR TITLE
feat: modal

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Move modal from NNS-dapp and update for new UI design.

# Changes

lib:

- move `Modal` cmp from NNS-dapp
- update new design
- replace property `size` with `role=dialog|alert`
- replace `footer` slot with `toolbar`

docs:

- document accordingly
- add new docs component to generate lorem ipsum paragraphs

# Screenshots

b

Capture d’écran 2022-09-13 à 14 26 54
Capture d’écran 2022-09-13 à 14 26 59
Capture d’écran 2022-09-13 à 14 27 08
Capture d’écran 2022-09-13 à 14 27 15

### CODE REVIEW

File: src/routes/components/modal/+page.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+| `role`                 | The modal is either a dialog meant for getting work done or an alert that requires immediate attention. | `dialog` or `alert`     | `dialog`    |
+| `testId`               | Add a `data-tid` attribute to the DOM, useful for test purpose.                                         | `string` or `undefined` | `undefined` |
+| `disablePointerEvents` | Disable interactive elements - close actions - of the modal.                                            | `boolean`               | `false`     |
+
+## Slots
+
+| Slot name    | Description                                                                                  |
+| ------------ | -------------------------------------------------------------------------------------------- |
+| Default slot | The content of the modal.                                                                    |
+| Title        | The title of the modal. Displayed in a toolbar with a "Close" icon button on the right side. |
```
Comment (mstrasinskis) [inline src/routes/components/modal/+page.md:43]: What do you think about using the lowercase version `title` to reflect the exact slot value?

**Review by mstrasinskis — APPROVED** [review]
LGTM, minor comment.

File: src/routes/components/modal/+page.md (hunk 1 above)
Comment (peterpeterparker) [reply src/routes/components/modal/+page.md:43]: Absolutely, that's a typo 😅

Comment (peterpeterparker) [reply src/routes/components/modal/+page.md]: Done.

**Pull request merged**